lightstep/sdk/metric/sdkinstrument/performance.go

```diff
@@ -81,6 +89,42 @@ func (p Performance) Validate() Performance {
 	if p.AggregatorCardinalityLimit == 0 {
 		p.AggregatorCardinalityLimit = DefaultAggregatorCardinalityLimit
 	}
+	if p.AttributeSizeLimit == 0 {
+		p.AttributeSizeLimit = DefaultAttributeSizeLimit
+	}
 
 	return p
 }
+
+func (p Performance) TruncateAttributes(attrs []attribute.KeyValue) []attribute.KeyValue {
+	sizeLimit := p.AttributeSizeLimit
+	if sizeLimit == 0 {
+		return attrs
+	}
+	for idx, kv := range attrs {
+		if len(kv.Key) > int(sizeLimit) {
+			attrs[idx].Key = kv.Key[:sizeLimit]
+		}
+		switch attrs[idx].Value.Type() {
```

Comment:
Do we care about truncating other attribute slice types like an int64slice, float64slice?

Reply:
It's a concern, but to truncate strings and string slices is an explicit part of the spec, whereas the other types would be more off-spec. I see the other types as less likely to create large values, so this seems a reasonable compromise.

https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/common#attribute-limits